Mathlib/RingTheory/LaurentSeries.lean

```diff
@@ -609,3 +617,169 @@ theorem val_le_one_iff_eq_coe (f : LaurentSeries K) : Valued.v f ≤ (1 : ℤₘ
 end LaurentSeries
 
 end AdicValuation
+namespace LaurentSeries
+section Complete
+
+open Filter
+
+open scoped DiscreteValuation
+
+variable {K : Type*} [Field K]
+
+/- Sending a Laurent series to its `d`-th coefficient is uniformly continuous when the coefficient
+field has the discrete uniformity. -/
+theorem uniformContinuous_coeff {uK : UniformSpace K} (h : uK = ⊥) (d : ℤ) :
+    UniformContinuous fun f : LaurentSeries K ↦ f.coeff d := by
+  refine uniformContinuous_iff_eventually.mpr fun S hS ↦ eventually_iff_exists_mem.mpr ?_
+  let γ : ℤₘ₀ˣ := Units.mk0 (↑(Multiplicative.ofAdd (-(d + 1)))) WithZero.coe_ne_zero
+  use {P | Valued.v (P.snd - P.fst) < ↑γ}
+  refine ⟨(Valued.hasBasis_uniformity (LaurentSeries K) ℤₘ₀).mem_of_mem (by tauto), fun P hP ↦ ?_⟩
+  rw [eq_coeff_of_valuation_sub_lt K (le_of_lt hP) (lt_add_one _)]
+  apply bot_uniformity ▸ h ▸ hS; rfl
+
+/-- Since extracting coefficients is uniformly continuous, every Cauchy filter in
+`laurent_series K` gives rise to a Cauchy filter in `K` for every `d : ℤ`, and such Cauchy filter
+in `K` converges to a principal filter -/
+def Cauchy.coeff {ℱ : Filter (LaurentSeries K)} (hℱ : Cauchy ℱ) : ℤ → K :=
+  let _ : UniformSpace K := ⊥
+  fun d ↦ UniformSpace.DiscreteUnif.cauchyConst rfl <| hℱ.map (uniformContinuous_coeff rfl d)
+
+theorem Cauchy.coeff_tendsto {ℱ : Filter (LaurentSeries K)} (hℱ : Cauchy ℱ) (D : ℤ) :
+    Tendsto (fun f : LaurentSeries K ↦ f.coeff D) ℱ (𝓟 {coeff hℱ D}) :=
+  let _ : UniformSpace K := ⊥
+  le_of_eq <| UniformSpace.DiscreteUnif.eq_const_of_cauchy (by rfl)
+    (hℱ.map (uniformContinuous_coeff rfl D)) ▸ (principal_singleton _).symm
+
+/- For every Cauchy filter of Laurent series, there is a `N` such that the `n`-th coefficient
+vanishes for all `n ≤ N` and almost all series in the filter. This is an auxiliary lemma used
+to construct the limit of the Cauchy filter as a Laurent series, ensuring that the support of the
+limit is `PWO`.-/
+lemma Cauchy.exists_lb_eventual_support {ℱ : Filter (LaurentSeries K)} (hℱ : Cauchy ℱ) :
```

Comment:
Here, I'm certain that the fact that one uses Laurent series is not needed, the same proof would probably show that Hahn series are complete.

Reply:
Well, my problem here is that I make strong use of an explicit description of the uniformity on `LaurentSeries K` coming from its `Valued` group structure, where entourages are defined in terms of the `X`-adic valuation. Nothing of this is currently available for Hahn series. I can of course make a remark (or you can add it to this PR, please do feel free!) but I fear that generalizing the result now would result in a gigantic refactor.

Reply:
Then a remark will be OK. Ultimately people will prove that the natural topology on Hahn series is complete, and that for Z, it gives the topology given by the valuation.